test/active_record_doctor/detectors/missing_unique_indexes_test.rb

```diff
@@ -109,6 +109,57 @@ def test_skips_validator_without_attributes
     refute_problems
   end
 
+  def test_has_one_without_index
```

Comment:
We also need three other test cases:

- `ignore_models` defined globally
- `ignore_models` defined on the detector only
- `ignore_columns` defined on the detector (there's no global setting)

You can have a look at other test cases for reference.